ludwig/models/llm.py

```diff
@@ -89,17 +91,35 @@ def __init__(
         self._random_seed = random_seed
 
         self.model_name = self.config_obj.base_model
+        self.model_config = AutoConfig.from_pretrained(self.config_obj.base_model)
+
+        self.is_ray_backend = is_ray_backend(self.config_obj.backend)
 
         logger.info("Loading large language model...")
-        self.model = AutoModelForCausalLM.from_pretrained(self.config_obj.base_model)
-        self.curr_device = torch.device("cpu")  # model initially loaded onto cpu
+        if self.is_ray_backend:
+            # Extract weights as numpy tensors and place them in the Ray object store.
+            # If we store the weights of a model as NumPy arrays on Plasma, we can access those
+            # weights directly out of Plasma’s shared memory segments, without making any copies.
+            # This enables zero copy model loading on each training worker using shared
+            # memory from the Ray object store for model initialization.
+            import ray
+
+            self.model_ref = ray.put(
```

Comment:
Why did we get rid of `self.model`? After the extract function is called, it has no data. So there's no reason not to assign it to `self.model`, as far as I can tell.

Reply:
I just felt like we didn't need to keep a copy in CPU memory at all at this point - are you suggesting the following change?

```
self.model = AutoModelForCausalLM.from_pretrained(self.config_obj.base_model)
self.model_ref = ray.put(extract_tensors(self.model))
```

And the idea is that self.model still has the skeleton, just that the underlying weights have been set to None in place? I think that would probably be ok!